src/BotwinExtensions.cs

```diff
@@ -54,6 +59,34 @@ public static IApplicationBuilder UseBotwin(this IApplicationBuilder builder, Bo
             return builder.UseRouter(routeBuilder.Build());
         }
 
+        private static void Apply405Handler(IApplicationBuilder builder, IEnumerable<BotwinModule> modules)
+        {
+            var systemRoutes = new List<(string verb, string route)>();
+            foreach (var module in modules)
+            {
+                foreach (var route in module.Routes.Keys)
+                {
```

Comment:
We’re already iterating the module routes in line 48, so we could just combine collecting all “system routes” with setting up the normal route handler here, to avoid iterating everything twice.

Reply:
I initially tried to do that but i need to loop to get all routes before applying the 405 handler which is global whereas the route handler is per route

Reply:
I was thinking about something like this:

```
foreach (var route in …)
{
    routeBuilder.MapVerb(…);
    knownRoutes.Add(route);
}

builder.UseRouter(routeBuilder.Build());
builder.Use(GetMethodNotAllowedHandler(knownRoutes));
```